lib/systemd/system/delphix-migration.service

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# We want to execute this service as early as possible since it will be
+# finalizing the network configuration after an OS migration. Since it will
+# also be importing domain0, which is normally imported by the
+# zfs-import-cache service, we add similar dependencies to it
+# (i.e. devices must be ready, so run after systemd-udev-settle.service, and
+# service is required by zfs-import.target). Finally, we must make sure to
+# remove default service dependencies, as those would create a dependency
+# cycle (this service must run before local-fs.target, but default
+# dependencies require local-fs.target).
+#
+[Unit]
+Description=Delphix OS Migration Service
+DefaultDependencies=no
+After=systemd-udev-settle.service
+Before=network-pre.target
+Before=zfs-import.target
```

Comment:
Does this need to run to completion, before the `delphix-platform` (and all the service that depend on it) run? If so, do we indirectly accomplish this via these statements?

Reply:
Yes and yes.
`zfs-import.target` is required by `zfs-mount.service` which is required by `local-fs.target`, which falls within the basic service dependencies.

Reply:
Can you leave a comment to explain this? This requirement isn't entirely clear as-is.